paimon-core/src/main/java/org/apache/paimon/mergetree/compact/aggregate/AggregateMergeFunction.java

```diff
@@ -63,11 +65,25 @@ public AggregateMergeFunction(
     public void reset() {
         this.latestKv = null;
         this.row = new GenericRow(getters.length);
+        this.initialKv = null;
+        this.isInitialized = false;
         Arrays.stream(aggregators).forEach(FieldAggregator::reset);
     }
 
     @Override
     public void add(KeyValue kv) {
+        if (initialKv == null) {
+            initialKv = kv;
```

Comment:
Is the reason for extracting `initialKv` separately to handle the case where there is only one kv and it is a retract? What if there are multiple kvs, but all of them are retracts? It seems like it will still return an INSERT record? It looks like this situation existed before?

Reply:
Yes, you are right.
If there are multiple kvs, but all of them are retracts, we still return as **INSERT** record now.
Do we need to deal with this situation?

Reply:
While, this looks like a bug to me, I remember that I previously submitted a related issue #4670.

I'm fine with not including the fix for this issue in this PR since it is beyond the scope of #4999.

@yuzelin What do you think? Can you help review? Thanks!